### PR TITLE
Cisdp 1308,10,19-23

### PR DESCRIPTION
**Pull request by slamothe99**
## Pre-review checklist

I have ensured that:

- [ ] Code runs successfully in the development environment.
- [ ] New code is tested to prove that it works. If not, justify reasoning.
- [ ] New code is documented using [type hinting](https://realpython.com/lessons/type-hinting/) and following the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) docstring format.
- [ ] New code follows the project naming standards.

## Post-review checklist

As I merge these changes, I have:

- [ ] Updated the related Jira ticket, including approximate time spent (including review).
- [ ] Moved the related Jira ticket to Done.

### CODE REVIEW

File: cishouseholds/derive.py [hunk 1]
```diff
@@ -810,14 +814,22 @@ def assign_date_difference(
         First date column name.
     end_reference_column
         Second date column name.
-
+    format
+        time format (days, weeks, months)
```
Comment (foster999) [inline cishouseholds/derive.py:818]: This suggests that the function supports `"months"` but there doesn't seem to be any functionality for that below. Please also add validation that the user has supplied a supported format, as the code below will treat `format="potato"` as `"days"`

**Review by foster999 — COMMENTED** [review]
Please could you give this a run in the dev environment, to make sure the union_dependent_transformation runs ok?

Other than these two comments, looks good to me

File: cishouseholds/derive.py (hunk 1 above)
Comment (slamothe99) [reply cishouseholds/derive.py:818]: added validation

Comment (slamothe99) [reply cishouseholds/derive.py:818]: > Please could you give this a run in the dev environment, to make sure the union_dependent_transformation runs ok?
> 
> Other than these two comments, looks good to me

we will create a new ticket for this as it is part of a larger bug

**Review by ohibo — APPROVED** [review]

**Pull request merged**